src/lit/youtube-inline.ts

```diff
@@ -0,0 +1,174 @@
+import {LitElement, css, html} from 'lit';
+import {query, property} from 'lit/decorators.js';
+import {TetrahedronBufferGeometry} from 'three';
+
+import {ScriptLoader} from '../loader/script-loader';
+const YOUTUBE_IFRAME_API = 'https://www.youtube.com/iframe_api?trustedtypes=1';
+
+/**
+ * # Degu Youtube Inline Component
+ * Automatically adds an youtube iframe to your page.
+ *
+ * Usage:
+ *
+ * ```ts
+ * import { DeguYoutubeInline} from '@blinkk/degu/lib/lit/youtube-inline';
+ *
+ * window.customElements.define('degu-youtube-inline', DeguYoutubeInline);
+ * ```
+ *
+ * ```html
+ *  <degu-youtube-inline
+ *    video-id="5qap5aO4i9A"
+ *    style="aspect-ratio: 16/9"></degu-youtube-inline>
+ * ```
+ *
+ *
+ * Advanced Example:
+ * Here is an example of programmatically updating the youtube video.
+ *
+ * ```html
+ *  <degu-youtube-inline id="myVideo"
+ *    style="aspect-ratio: 16/9"></degu-youtube-inline>
+ *
+ *   <button id="change-youtube-video">Change Video</button>
+ * ```
+ *
+ * ```ts
+ * import { DeguYoutubeInline} from '@blinkk/degu/lib/lit/youtube-inline';
+ * window.customElements.define('degu-youtube-inline', DeguYoutubeInline);
+ *
+ *   // Initially load a video
+ *   const youtubeInlineTest = document.getElementById('myVideo');
+ *   youtubeInlineTest.load('t_cKM_JYtbs')
+ *
+ *    let index = 0;
+ *    const youtubeIds = [
+ *      '_tV5LEBDs7w',
+ *      'hGIW2fDb0jg',
+ *      '4pcNRDx6KrE',
+ *      'UtWaBEn_x8c'
+ *    ]
+ *
+ *    // On button clicks change the youtube video.
+ *    document.getElementById('change-youtube-video').addEventListener('click', ()=> {
+ *      index++;
+ *      if(index >= youtubeIds.length - 1) {
+ *        index = 0;
+ *      }
+ *      youtubeInlineTest.load(youtubeIds[index], true)
+ *    })
+ * ```
+ *
+ */
+export class DeguYoutubeInline extends LitElement {
```

Comment:
We should be consistent. I think that since the official product name is "YouTube" we may want to go with "YouTube" to be consistent with that. But, the elements should be "youtube" not "you-tube". Does that work?

Reply:
Done